### ISSUE TITLE
improve performance by making Roblox API requests async and running them in parallel when possible

### ISSUE THREAD

**Issue by blake-mealey** [blake-mealey/mantle#18]
Since #16, Tokio has been added for making asynchronous requests to AWS. We can use this to improve the performance of deploys by parallelizing Roblox API requests. We will have to run the requests in batches and only run the next N requests which are not dependent on each other.

**Comment by blake-mealey** [comment]
Consider fixing #89 when working on this

**Comment by blake-mealey** [comment]
Switch from `ureq` to `reqwest` to implement this. Also look into creating a reusable HTTP client and saving cookies in them using the built-in session store rather than handling them manually.

**Comment by blake-mealey** [comment]
Closing this as it is really not a priority (performance is great anyways) and would require significantly more complex and less readable core resolver logic.
